apps/opik-frontend/src/components/pages-shared/automations/RunEvaluationDialog/RunEvaluationDialog.tsx

```diff
@@ -0,0 +1,320 @@
+> = ({ open, setOpen, projectId, entityIds, entityType }) => {
+  const workspaceName = useAppStore((state) => state.activeWorkspaceName);
+  const [selectedRuleIds, setSelectedRuleIds] = useState<Set<string>>(
+    new Set(),
+  );
+  const [expandedRuleIds, setExpandedRuleIds] = useState<Set<string>>(
+    new Set(),
+  );
+  const [openCreateRuleDialog, setOpenCreateRuleDialog] =
+    useState<boolean>(false);
+
+  const { data, isLoading } = useRulesList(
+    {
+      workspaceName,
+      projectId,
+      page: 1,
+      size: 1000, // Load all rules for the project
+    },
+    {
+      enabled: open,
+      placeholderData: keepPreviousData,
+      staleTime: STALE_TIME,
+    },
+  );
+
+  const manualEvaluationMutation = useManualEvaluationMutation();
+
+  // Filter rules based on entity type
+  // Trace rules: llm_as_judge, user_defined_metric_python
+  // Thread rules: trace_thread_llm_as_judge, trace_thread_user_defined_metric_python
+  const rules = useMemo(() => {
+    const allRules = data?.content || [];
+
+    if (entityType === "trace") {
+      return allRules.filter(
+        (rule) =>
+          rule.type === EVALUATORS_RULE_TYPE.llm_judge ||
+          rule.type === EVALUATORS_RULE_TYPE.python_code,
+      );
+    } else if (entityType === "thread") {
+      return allRules.filter(
+        (rule) =>
+          rule.type === EVALUATORS_RULE_TYPE.thread_llm_judge ||
+          rule.type === EVALUATORS_RULE_TYPE.thread_python_code,
+      );
+    } else {
+      throw new Error(`Unknown entity type: ${entityType}`);
+    }
+  }, [data?.content, entityType]);
+
+  // Pre-select all rules when dialog opens and rules are loaded
+  useEffect(() => {
+    if (open && rules.length > 0 && selectedRuleIds.size === 0) {
+      setSelectedRuleIds(new Set(rules.map((rule) => rule.id)));
+    }
+  }, [open, rules, selectedRuleIds.size]);
+
+  const handleCheckboxChange = useCallback((ruleId: string) => {
+    setSelectedRuleIds((prev) => {
+      const newSet = new Set(prev);
+      if (newSet.has(ruleId)) {
+        newSet.delete(ruleId);
+      } else {
+        newSet.add(ruleId);
+      }
+      return newSet;
+    });
+  }, []);
+
+  const toggleExpanded = useCallback((ruleId: string) => {
+    setExpandedRuleIds((prev) => {
+      const newSet = new Set(prev);
+      if (newSet.has(ruleId)) {
+        newSet.delete(ruleId);
+      } else {
+        newSet.add(ruleId);
+      }
+      return newSet;
+    });
+  }, []);
+
+  const handleRunEvaluation = useCallback(() => {
+    if (selectedRuleIds.size === 0) return;
+
+    manualEvaluationMutation.mutate(
+      {
+        projectId,
+        entityIds,
+        ruleIds: Array.from(selectedRuleIds),
+        entityType,
+      },
+      {
+        onSuccess: () => {
+          setOpen(false);
+          setSelectedRuleIds(new Set());
+          setExpandedRuleIds(new Set());
+        },
+      },
+    );
+  }, [
+    selectedRuleIds,
+    manualEvaluationMutation,
```

Comment:
you should avoid sending `manualEvaluationMutation` as a  full object to `useEffect` dependency, replace it with `manualEvaluationMutation.mutate`